scripts/pi-hole/php/savesettings.php

```diff
@@ -635,7 +635,19 @@ function readAdlists()
 
 				// Reread available adlists
 				$adlist = readAdlists();
+				break;
 
+			case "privacyLevel":
+				$level = intval($_POST["privacylevel"]);
+				if($level >= 0 && $level <= 3)
+				{
+					exec("sudo pihole -a privacylevel ".$level);
+					$success .= "The privacy level has been updated";
+				}
+				else
+				{
+					$error .= "Invalid privacy level (".$level.")!";
```

Comment:
This output should be sanitized.

Reply:
I don't think this is needed. `$level` is the returned value of `intval()` and hence guaranteed to be an integer: 

> `int intval ( mixed $var [, int $base = 10 ] )`
> The integer value of `var` on success, or 0 on failure.

http://php.net/manual/en/function.intval.php